source/MaterialXRender/Util.cpp

```diff
@@ -0,0 +1,228 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRender/Util.h>
+
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+
+namespace MaterialX
+{
+
+ShaderPtr createShader(const string& shaderName, GenContext& context, ElementPtr elem)
+{
+    if (!elem)
+    {
+        return nullptr;
+    }
+
+    context.getOptions().hwTransparency = isTransparentSurface(elem, context.getShaderGenerator());
+    return context.getShaderGenerator().generate(shaderName, elem, context);
+}
+
+ShaderPtr createConstantShader(GenContext& context,
+                            DocumentPtr stdLib,
+                            const string& shaderName,
+                            const Color3& color)
+{
+    // Construct the constant color nodegraph
+    DocumentPtr doc = createDocument();
+    doc->importLibrary(stdLib);
+    NodeGraphPtr nodeGraph = doc->addNodeGraph();
+    NodePtr constant = nodeGraph->addNode("constant");
+    constant->setParameterValue("value", color);
+    OutputPtr output = nodeGraph->addOutput();
+    output->setConnectedNode(constant);
+
+    // Generate the shader
+    ShaderPtr shader = createShader(shaderName, context, output);
+    return shader;
+}
+
+VariableBlock* getUniformBlock(ShaderPtr shader, const string& stageName, const string& blockName)
+{
+    try
+    {
+        ShaderStage& stage = shader->getStage(stageName);
+        VariableBlock& block = stage.getUniformBlock(blockName);
+        return &block;
+    }
+    catch (Exception& /*e*/)
+    {
+    }
+    return nullptr;
+}
+
+ShaderPort* findUniform(const VariableBlock* block, const std::string& path)
```

Comment:
I think the same goes for this function, it's also used in just a single place, as far as I can see. For this one I agree it might be useful in other places later, but until that happens maybe we should wait adding it as a utility (if you are not certain already).

If you want to keep it the naming of the function is not entirely correct. The block given could be any block not just for uniforms, since our VariableBlock class is used for inputs/varyings and outputs as well. So maybe `findVariable` is better.

I think it would be better to add a `getVariable("path")` method on the VariableBlock class. Or even higher up on the Stage or Shader class we could have a find-variable-by-path method, which also would use a map to do the search quicker than linear search. This is probably for another change list though :)

Reply:
I think we can get rid of the the getUniformBlock(). For this one, I like the idea of adding VariableBlock::getVariable("path"). It may be a linear search for now which can be optimized if we need to.